wait-task/pkg/reconciler/reconciler_test.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2021 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reconciler
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/apis"
+)
+
+var validRef = &v1alpha1.TaskRef{
+	APIVersion: "example.dev/v0",
+	Kind:       "Wait",
+}
+
+func TestReconcile(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	r := &v1alpha1.Run{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "run",
+		},
+		Spec: v1alpha1.RunSpec{
+			Ref: validRef,
+			Params: []v1beta1.Param{{
+				Name:  "duration",
+				Value: *v1beta1.NewArrayOrString("1s"),
+			}},
+		},
+	}
+	rec := &Reconciler{}
+	// Mock the EnqueueAfter method to sleep, then call ReconcileKind
+	// again. This will get called after ReconcileKind is called the first
+	// time, below. After reconciling again, check that the status is as
+	// expected.
+	rec.EnqueueAfter = func(_ interface{}, d time.Duration) {
```

Comment:
If I didn't set it at all, it would nil panic when it tried to call it. If I made it a no-op, the first `ReconcileKind` would call it and return, and nothing would take responsibility for re-enqueueing it after the delay.

I could probably hook this up to a mock controller with a mock workqueue, but I don't think it's really worth it at this point.